src/core/request.js

```diff
@@ -86,6 +86,7 @@ const createError = (response, fallbackMessage, httpCode, httpSent) => {
  * @param {Boolean} [options.background] - if true, the request should be done in the background, which in practice does not trigger the global handlers like ajaxStart or ajaxStop
  * @param {Boolean} [options.sequential] - if true, the request must join a queue to be run sequentially
  * @param {Number}  [options.timeout] - timeout in seconds for the AJAX request
+ * @param {Object} [options.jwtTokenHandler] - JWT token handler instance
```

Comment:
When I talked with JS about the implementation plan, he said I should support multiple services. Because of that the tokenHandler cannot be integrated into the request module, so it should be provided. If we should support only one service then the easiest way is to create a singleton module that will call tokenHandlerFactory, have an instance and it could return with this instance every time when it is necessary. The integration into nextgen was not 100% clear for me so I created a common implementation. Let me know if you do not like it and I could modify.

Reply:
ok then. maybe we should have handled the `csrfTokenHandler` the same way. My only concern is the consistency